src/Octoshift/RetryPolicy.cs

```diff
@@ -41,23 +41,27 @@ public async Task<PolicyResult<T>> RetryOnResult<T>(Func<Task<T>> func, T result
         public async Task Retry(Func<Task> func)
         {
             var policy = Policy.Handle<Exception>()
-                               .WaitAndRetryAsync(5, retry => retry * TimeSpan.FromMilliseconds(_retryInterval), (_, _) =>
+                               .WaitAndRetryAsync(5, retry => retry * TimeSpan.FromMilliseconds(_retryInterval), (ex, _) =>
                                {
+                                   _log?.LogVerbose(ex is HttpRequestException httpEx ? $"[HTTP ERROR {(int?)httpEx.StatusCode}] {ex}" : ex.ToString());
                                    _log?.LogVerbose("Retrying...");
                                });
 
             await policy.ExecuteAsync(func);
         }
 
-        public async Task<PolicyResult<T>> Retry<T>(Func<Task<T>> func)
+        public async Task<T> Retry<T>(Func<Task<T>> func)
         {
             var policy = Policy.Handle<Exception>()
                                .WaitAndRetryAsync(5, retry => retry * TimeSpan.FromMilliseconds(_retryInterval), (ex, _) =>
                                {
-                                   _log?.LogVerbose($"Failed with {ex.GetType().Name}. Retrying...");
+                                   _log?.LogVerbose(ex is HttpRequestException httpEx ? $"[HTTP ERROR {(int?)httpEx.StatusCode}] {ex}" : ex.ToString());
+                                   _log?.LogVerbose("Retrying...");
                                });
 
-            return await policy.ExecuteAndCaptureAsync(func);
+            var result = await policy.ExecuteAndCaptureAsync(func);
+
+            return result.Outcome == OutcomeType.Failure ? throw result.FinalException : result.Result;
```

Comment:
Good idea to get rid of the capturing retry (I also made a similar change in my local branch). I just have to suggestions:
1. There is no need to call `ExecuteAndCaptureAsync` in the second `Retry` it can simply just be `ExecuteAsync`. 
2. To be a bit more DRY, we can create a private method to build/create the policy and then call it in both `Retry` methods, like this:
```c#
    public async Task Retry(Func<Task> func) => await CreateRetryPolicyForException<Exception>().ExecuteAsync(func);

    public async Task<T> Retry<T>(Func<Task<T>> func) => await CreateRetryPolicyForException<Exception>().ExecuteAsync(func);

    private AsyncRetryPolicy CreateRetryPolicyForException<TException>() where TException : Exception => Policy
            .Handle<TException>()
            .WaitAndRetryAsync(5, retry => retry * TimeSpan.FromMilliseconds(_retryInterval), (ex, _) =>
            {
                _log?.LogVerbose(ex is HttpRequestException httpEx ? $"[HTTP ERROR {(int?)httpEx.StatusCode}] {ex}" : ex.ToString());
                _log?.LogVerbose("Retrying...");
            });
```

Reply:
good suggestion, done!